backend/tests/unit/graphql/test_graphql_query.py

```diff
@@ -1535,6 +1535,74 @@ async def test_query_filter_relationship_id(
     assert len(result.data["TestPerson"]["edges"][0]["node"]["cars"]["edges"]) == 2
 
 
+@pytest.mark.parametrize(
+    "graphql_filter,expected_results",
+    [
+        ('mylist__value: "tree"', ["obj1", "obj2"]),
+        ("mylist__value: 2", ["obj2", "obj3", "obj5"]),
+        ('mylist__value: "one", level__value: 2', ["obj2"]),
+        ('mylist__values: ["one"]', ["obj1", "obj2", "obj3"]),
+        ('mylist__values: ["one", "two"]', ["obj1", "obj2", "obj3", "obj5"]),
+        ('mylist__values: ["one", "two"]', ["obj1", "obj2", "obj3", "obj5"]),
+        ('mylist__values: ["one", 5]', ["obj1", "obj2", "obj3", "obj4"]),
+        ("mylist__value: true", ["obj3"]),
+        ("mylist__values: [true]", ["obj3"]),
+        ("mylist__values: [true, false]", ["obj3", "obj5"]),
+    ],
+)
+async def test_query_filter_list(
+    db: InfrahubDatabase,
+    default_branch: Branch,
+    criticality_schema: NodeSchema,
+    graphql_filter: str,
+    expected_results: list[str],
+):
+    obj1 = await Node.init(db=db, schema=criticality_schema)
+    await obj1.new(db=db, name="obj1", level=1, mylist=["one", "two", "tree", 5])
+    await obj1.save(db=db)
+    obj2 = await Node.init(db=db, schema=criticality_schema)
+    await obj2.new(db=db, name="obj2", level=2, mylist=["one", 2, "tree"])
+    await obj2.save(db=db)
+    obj3 = await Node.init(db=db, schema=criticality_schema)
+    await obj3.new(db=db, name="obj3", level=3, mylist=["one", "two", True, 2])
+    await obj3.save(db=db)
+    obj4 = await Node.init(db=db, schema=criticality_schema)
+    await obj4.new(db=db, name="obj4", level=4, mylist=["anotherone", "twotree", "true", "2", 5])
+    await obj4.save(db=db)
+    obj5 = await Node.init(db=db, schema=criticality_schema)
+    await obj5.new(db=db, name="obj5", level=5, mylist=["oneone", "two", False, 2])
```

Comment:
Some of these entries feels like they might work due to a bug. Do we want to support having a list of different type of objects?

Reply:
It's not idea but currently we technically support having a list with different types of data
Once #4246 and #3931 have been implemented it will be possible to restrict that

The good part is for the query and for this fix we can easily support both